libsolidity/parsing/Parser.cpp

```diff
@@ -406,6 +406,8 @@ ASTPointer<EnumDefinition> Parser::parseEnumDefinition()
 		if (m_scanner->currentToken() != Token::Identifier)
 			fatalParserError(string("Expected Identifier after ','"));
 	}
+	if (members.size() == 0)
+		fatalParserError({"enum with no members is not allowed."});
```

Comment:
Parser errors only have to be fatal if parsing cannot continue, which is not the case here, I would say.